tests/test_trading_calendar.py

```diff
@@ -575,9 +575,31 @@ def _get_session_block(self):
         # session) block.
 
         if not self.HAVE_EARLY_CLOSES:
-            # If we don't have any early closes, just return a "random" chunk
-            # of three sessions.
-            return self.calendar.all_sessions[10:13]
+            # If we don't have any early closes, return a "random" chunk of
+            # three sessions. To avoid any edge cases, make sure the first and
+            # last sessions are adjacent to days that are also sessions.
+            one_day = pd.Timedelta('1 Day')
+            all_sessions = self.calendar.all_sessions
+            found_enclosed_block = False
+            i = 0
+            while not found_enclosed_block:
+                i += 1
+                try:
+                    block_start = all_sessions[i]
+                    block_end = all_sessions[i + 2]
+                    pre_session = all_sessions[i - 1]
+                    post_session = all_sessions[i + 3]
+                except IndexError:
+                    raise AssertionError(
+                        f'Failed to find a suitable block of test sessions '
```

Comment:
We're definitely not going to be able to support f-strings in this repo any time soon.